features/src/main/feature/camel-features.xml

```diff
@@ -1554,14 +1554,13 @@
 <!--        <feature version='${project.version}'>camel-core</feature>-->
 <!--        <bundle>mvn:org.apache.camel/camel-quartz/${project.version}</bundle>-->
 <!--    </feature>-->
-<!--    <feature name='camel-quickfix' version='${project.version}' start-level='50'>-->
-<!--        <feature version='${project.version}'>camel-core</feature>-->
-<!--        <bundle dependency='true'>mvn:org.apache.mina/mina-core/${mina-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.xmlresolver/${xmlresolver-bundle-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.saxon/${saxon-bundle-version}</bundle>-->
-<!--        <bundle>mvn:org.quickfixj/quickfixj-all/${quickfixj-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-quickfix/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-quickfix' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <bundle dependency="true">wrap:mvn:net.sf.saxon/Saxon-HE/12.4</bundle>
+        <bundle dependency='true'>mvn:org.apache.mina/mina-core/${mina-version}</bundle>
+        <bundle dependency='true'>mvn:org.quickfixj/quickfixj-all/${quickfixj-version}</bundle>
```

Comment:
If you check well in the camel project, you will see that only `quickfixj-core` and `quickfixj-messages-all` are used, consequently saxon is no more needed

Reply:
We need to avoid being too influenced by the previous code

Reply:
Saxon is indeed absent in the dependency tree
but if try to build with it
`[ERROR] Message: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=camel-quickfix; type=karaf.feature; version=4.5.0.SNAPSHOT; filter:="(&(osgi.identity=camel-quickfix)(type=karaf.feature)(version>=4.5.0.SNAPSHOT))" [caused by: Unable to resolve camel-quickfix/4.5.0.SNAPSHOT: missing requirement [camel-quickfix/4.5.0.SNAPSHOT] osgi.identity; osgi.identity=camel-quickfix; type=osgi.bundle; version="[4.5.0.SNAPSHOT,4.5.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve camel-quickfix/4.5.0.SNAPSHOT: missing requirement [camel-quickfix/4.5.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.quickfixj.jmx)(version>=2.3.0)(!(version>=3.0.0)))" [caused by: Unable to resolve org.quickfixj.all/2.3.1: missing requirement [org.quickfixj.all/2.3.1] osgi.wiring.package; filter:="(osgi.wiring.package=net.sf.saxon)"]]]`

quickfix-all imports `net.sf.saxon` in its manifest
should I use wrap and override the imports?

Reply:
no, you should add `quickfixj-core` and `quickfixj-messages-all` instead of `quickfixj-all` like in the pom file of the corresponding camel component

Reply:
fixed, thanks